frontend/scripts/gameLogicInterface.js

```diff
@@ -226,6 +226,13 @@ export default function createGame(initialGameState = emptyGameState) {
 		holdCurrentTetromino: function() {
 
 		},
+		updateActiveTetromino: function() {
+			this.gameState.activeTetromino.name = this.gameState.upcomingTetrominoes.shift()
+				this.gameState.activeTetromino.tiles = TetrominoShapes[this.gameState.activeTetromino.name]

```

Comment:
Take a look at how the activeTetromino is defined in the original `emptyGameState` - at the moment all the new tiles will be the same colour! (And possibly the same position as the old tile)

```
	activeTetromino: {
		...(function() {
			const tetromino = getRandomTetromino();
			return {
				name: tetromino,
				tiles: TetrominoShapes[tetromino]
			};
		}()),
		position: {
			x: (BOARD_UNITS_WIDTH - 4) / 2,
			y: BOARD_UNITS_HEIGHT - 1, // Top row is reserved for game over
		},
		colour: getRandomColour()
	}
```

Reply:
You might want to consider refactoring (you might need to look up what this word means!) this code so that you can call it in both places